lib/SimpleSAML/Utils/EMail.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace SimpleSAML\Utils;
+
+use PHPMailer\PHPMailer\PHPMailer;
+use PHPMailer\PHPMailer\Exception;
+
+use SimpleSAML\Configuration;
+use SimpleSAML\Logger;
+use SimpleSAML\XHTML\Template;
+
+/**
+ * E-mailer class that can generate a formatted e-mail from array
+ * input data.
+ *
+ * @author Jørn Åne de Jong, Uninett AS <jorn.dejong@uninett.no>
+ * @package SimpleSAMLphp
+ */
+
+class EMail
+{
+
+    /**
+     * Get the default e-mail address from the configuration
+     * This is used both as source and destination address
+     * unless something else is provided at the constructor.
+     *
+     * It will refuse to return the SimpleSAMLphp default address,
+     * which is na@example.org.
+     *
+     * @return string Default mail address
+     */
+    public static function getDefaultMailAddress() {
+        $config = Configuration::getInstance();
+        $address = $config->getString('technicalcontact_email', 'na@example.org');
+        if ('na@example.org' === $address) {
+            throw new \Exception('technicalcontact_email not set');
+        }
+        return $address;
+    }
+
+    /** @var array Dictionary with multivalues */
+    private $data;
+    /** @var string Introduction text */
+    private $text;
+    /** @var PHPMailer The mailer instance */
+    private $mail;
+
+    /**
+     * Constructor
+     *
+     * @throws PHPMailer\PHPMailer\Exception;
+     */
+    public function __construct($subject, $from = null, $to = null) {
+        $this->mail = new PHPMailer(true);
+        $this->mail->Subject = $subject;
+        $this->mail->setFrom($from ? $from : static::getDefaultMailAddress());
+        $this->mail->addAddress($to ? $to : static::getDefaultMailAddress());
+    }
+
+    /**
+     * Set the data that should be embedded in the e-mail body
+     *
+     * @param array $data The data that should be embedded in the e-mail body
+     */
+    public function setData($data) {
+        $this->data = array_map(function($v){return is_array($v) ? $v : [$v];}, $data);
```

Comment:
this is not completely transparent code, not a problem but the comment could be a bit more explicit as to what it does